src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -1474,6 +1476,47 @@ protected override void InvokeDefaultAction(string path)
             return true;
         }
 
+        //Returns true if a given DirectoryInfo path is a recursive sym link.
+        private bool IsRecursiveSymLink(DirectoryInfo path)
+        {
+            string readSymLink = "";
+
+            if (Platform.IsWindows)
+            {
+                PSObject obj = PSObject.AsPSObject(path);
+
+                try
+                {
+                    readSymLink = InternalSymbolicLinkLinkCodeMethods.GetTarget(obj).First().ToString();
+                }
+                catch
+                {
+                    readSymLink = null;
+                }
+
+                //if the read symlink is not null or empty, and what it points to is the parent directory then it is recursive
+                if (!String.IsNullOrEmpty(readSymLink) && path.Parent.FullName.Contains(readSymLink))
+                {
+                    return true;
+                }
+
+                return false;
+            }
+
+            //Platform is Linux or Windows
+            else
+            {
+                readSymLink = Platform.NonWindowsInternalGetTarget(path.FullName);
```

Comment:
you catch all exception for win-plat above, but not for unix-plat. I think maybe it's better to not catch exception when calling into `GetTarget`. It shouldn't throw any exception normally, but if anything went wrong, it's probably better to let the user know.